compiler/AST/ParamForLoop.cpp

```diff
@@ -61,7 +61,35 @@ BlockStmt* ParamForLoop::buildParamForLoop(VarSymbol* indexVar,
     stride = new SymExpr(new_IntSymbol(1));
   }
 
-  if (call && call->isNamed("chpl_build_bounded_range"))
+  if (call && call->isNamed("#"))
+  {
+    Expr* count = new CallExpr("chpl_compute_count_param_loop", call->get(2)->remove());
+    call        = toCallExpr(call->get(1));
+    if (call && call->isNamed("chpl_build_low_bounded_range"))
+    {
+      low  = call->get(1)->remove();
+      high = new CallExpr("chpl_high_bound_count_for_param_loop", low->copy(), count);
+    }
+    else if (call && call->isNamed("chpl_build_high_bounded_range"))
+    {
+      high = call->get(1)->remove();
+      low  = new CallExpr("chpl_low_bound_count_for_param_loop", high->copy(), count);
+    }
+    else if (call && call->isNamed("chpl_build_bounded_range"))
+    {
+      Expr* temp_low = call->get(1);
+      Expr* temp_high = call->get(2);
+
+      // It is necessary that low is calculated first, because it also applies check for bound size.
+      low = new CallExpr("chpl_bounded_count_for_param_loop_low", temp_low->copy(), temp_high->copy(), count->copy());
+      high = new CallExpr("chpl_bounded_count_for_param_loop_high", temp_low->copy(), temp_high->copy(), count);
+    }
+    else
+    {
+      USR_FATAL(range, "iterators for param-for-loops must be bounded literal ranges");
```

Comment:
Given that `lo..#count` isn't exactly a bounded literal range, let's change this wording to something like "param for-loops currently only support literal range expressions with well-defined bounds?"

Reply:
sure sir.

Reply:
And line 99 should be similarly updated, I suppose.  Could the control flow here be refactored so that the error message is generated in a single place rather than in two distinct places, to avoid having to manually keep the error messages in synch?

Reply:
Yes, sir sure. I will do it.